# Introduction

...

In [1]:
# libraries to install
# !pip install selenium

In [2]:
import requests
from bs4 import BeautifulSoup
from tqdm import tqdm # useful for progress bars
from selenium import webdriver
import pandas as pd
import numpy as np
import time
import re
import os 

## 1. Data Collection

...

### 1.1. Get the list of animes

Here we will extract the *urls* and the *names* of the animes in the list (cfr. https://myanimelist.net/topanime.php). At first we can have an idea of the necessary steps to extract the informations we want by working on a single anime in the list and then proceed by iteration. 

After inspecting the HTML code of the site, we saw that the all the informations we need from a single anime are stored in  `tr` blocks inside a single `table` that contains the list of all the top animes in the site. To get the  name of an anime in the list we should work on `a` tags, whereas to get the url we need to work on `td` tags (leveraging the property `href`). 

Knowing these HTML details we can use the `selenium` library to do the web-scrapping.

In [3]:
from selenium.webdriver.chrome.service import Service

In [4]:
s = Service('/Users/dany/Desktop/adm-hw3/chromedriver')

In [5]:
# selenium with Chrome
driver = webdriver.Chrome(service=s)

In [6]:
# create a dataframe with links of each anime
df = pd.DataFrame(columns = ['Href'])

The following code was inspired by looking at the work that was done last year about https://www.goodreads.com, for example by https://github.com/GiorgiaSalvatori/ADM-HW3/blob/main/main.ipynb. Also the following post was useful https://towardsdatascience.com/how-to-use-selenium-to-web-scrape-with-example-80f9b23a843a.

In [7]:
# go page by page and and store links in a list
anime_list = []

for page in tqdm(range(0, 400)):
    url = 'https://myanimelist.net/topanime.php?limit=' + str(page * 50)
    response = requests.get(url)
    soup = BeautifulSoup(response.text, 'html.parser')
    for tag in soup.find_all('tr'):
        links = tag.find_all('a')
        for link in links:        
            if type(link.get('id')) == str and len(link.contents[0]) > 1:
                anime_list.append((link.contents[0], link.get('href')) )

100%|██████████| 400/400 [07:16<00:00,  1.09s/it]


In [8]:
# total number of animes
print(len(anime_list))

19124


In [9]:
# assign list to dataframe
df['Href'] = anime_list

In [10]:
# check for duplicates
df['Href'].nunique()

19124

It seems there is just one duplicate among the 19124 animes in the list. We think that a single repetition will not influence our analysis, so we leave the dataframe as it is.

In [11]:
# save dataframe into a csv file without header and comma separator. 
# this is equivalent to a txt file, but with also the names of the animes
# that can be of help in some data processing stages. 
df.to_csv('urls.csv',sep = ' ', header=False)

In [12]:
df['Href'].head()

0    (Fullmetal Alchemist: Brotherhood, https://mya...
1    (Gintama°, https://myanimelist.net/anime/28977...
2    (Shingeki no Kyojin Season 3 Part 2, https://m...
3    (Steins;Gate, https://myanimelist.net/anime/92...
4    (Fruits Basket: The Final, https://myanimelist...
Name: Href, dtype: object

We could also create a dictionary, this is useful in some circumnstances.

In [13]:
#keys
name = []   
#values
url = []    

for item in anime_list:
    name.append(item[0])
    url.append(item[1])
    
D = dict(zip(name, url))
print(len(list(D.values())[0:5]))

5


## 1.2. Crawl animes

We procede to:
- download the html corresponding to each of the collected urls;
- save its html in a file;
- organize the entire set of downloaded html pages into folders. Each folder will contain the htmls of the animes in page 1, page 2, ... of the list of animes.

To do so we extensively use the `os` library to create directories, changing paths, etc...

In [14]:
# remark: the execution can take quite some time

# returns current working directory
base = os.getcwd()  
t = 0
# we use the previously created dictionary D to get the urls we need
scrapped_urls = list(D.values())[0:-1]
for i in range(len(scrapped_urls)):
    if(i%50==0):
        # create a new folder
        # remark: the pages will start from 0
        page_identifier = i-(49*t)
        # subdirectory
        directory = f"page_{page_identifier}.html"
        # parent directories
        parent_dir = base
        # path
        path = os.path.join(parent_dir, directory)
        # make directory
        os.makedirs(path)
        # checkpoint
        print("Directory '%s' created" %directory)
        # change directory 
        os.chdir(path)
        t += 1
        
    # get urls
    URL = scrapped_urls[i]
    page = requests.get(URL)
    
    # parsing
    soup_data = BeautifulSoup(page.content, "html.parser")
    
    # saving
    with open(f"article_{i}.html", "w") as file:
        file.write(str(soup_data))
        
    # checkpoint
    print(f"Article {i} successfully written!")

Directory 'page_0.html' created
Article 0 successfully written!
Article 1 successfully written!
Article 2 successfully written!
Article 3 successfully written!
Article 4 successfully written!
Article 5 successfully written!
Article 6 successfully written!
Article 7 successfully written!
Article 8 successfully written!
Article 9 successfully written!
Article 10 successfully written!
Article 11 successfully written!
Article 12 successfully written!
Article 13 successfully written!
Article 14 successfully written!
Article 15 successfully written!
Article 16 successfully written!
Article 17 successfully written!
Article 18 successfully written!
Article 19 successfully written!
Article 20 successfully written!
Article 21 successfully written!
Article 22 successfully written!
Article 23 successfully written!
Article 24 successfully written!
Article 25 successfully written!
Article 26 successfully written!
Article 27 successfully written!
Article 28 successfully written!
Article 29 successful

Article 240 successfully written!
Article 241 successfully written!
Article 242 successfully written!
Article 243 successfully written!
Article 244 successfully written!
Article 245 successfully written!
Article 246 successfully written!
Article 247 successfully written!
Article 248 successfully written!
Article 249 successfully written!
Directory 'page_5.html' created
Article 250 successfully written!
Article 251 successfully written!
Article 252 successfully written!
Article 253 successfully written!
Article 254 successfully written!
Article 255 successfully written!
Article 256 successfully written!
Article 257 successfully written!
Article 258 successfully written!
Article 259 successfully written!
Article 260 successfully written!
Article 261 successfully written!
Article 262 successfully written!
Article 263 successfully written!
Article 264 successfully written!
Article 265 successfully written!
Article 266 successfully written!
Article 267 successfully written!
Article 268 succ

Article 478 successfully written!
Article 479 successfully written!
Article 480 successfully written!
Article 481 successfully written!
Article 482 successfully written!
Article 483 successfully written!
Article 484 successfully written!
Article 485 successfully written!
Article 486 successfully written!
Article 487 successfully written!
Article 488 successfully written!
Article 489 successfully written!
Article 490 successfully written!
Article 491 successfully written!
Article 492 successfully written!
Article 493 successfully written!
Article 494 successfully written!
Article 495 successfully written!
Article 496 successfully written!
Article 497 successfully written!
Article 498 successfully written!
Article 499 successfully written!
Directory 'page_10.html' created
Article 500 successfully written!
Article 501 successfully written!
Article 502 successfully written!
Article 503 successfully written!
Article 504 successfully written!
Article 505 successfully written!
Article 506 suc

Article 715 successfully written!
Article 716 successfully written!
Article 717 successfully written!
Article 718 successfully written!
Article 719 successfully written!
Article 720 successfully written!
Article 721 successfully written!
Article 722 successfully written!
Article 723 successfully written!
Article 724 successfully written!
Article 725 successfully written!
Article 726 successfully written!
Article 727 successfully written!
Article 728 successfully written!
Article 729 successfully written!
Article 730 successfully written!
Article 731 successfully written!
Article 732 successfully written!
Article 733 successfully written!
Article 734 successfully written!
Article 735 successfully written!
Article 736 successfully written!
Article 737 successfully written!
Article 738 successfully written!
Article 739 successfully written!
Article 740 successfully written!
Article 741 successfully written!
Article 742 successfully written!
Article 743 successfully written!
Article 744 su

Article 953 successfully written!
Article 954 successfully written!
Article 955 successfully written!
Article 956 successfully written!
Article 957 successfully written!
Article 958 successfully written!
Article 959 successfully written!
Article 960 successfully written!
Article 961 successfully written!
Article 962 successfully written!
Article 963 successfully written!
Article 964 successfully written!
Article 965 successfully written!
Article 966 successfully written!
Article 967 successfully written!
Article 968 successfully written!
Article 969 successfully written!
Article 970 successfully written!
Article 971 successfully written!
Article 972 successfully written!
Article 973 successfully written!
Article 974 successfully written!
Article 975 successfully written!
Article 976 successfully written!
Article 977 successfully written!
Article 978 successfully written!
Article 979 successfully written!
Article 980 successfully written!
Article 981 successfully written!
Article 982 su

Article 1185 successfully written!
Article 1186 successfully written!
Article 1187 successfully written!
Article 1188 successfully written!
Article 1189 successfully written!
Article 1190 successfully written!
Article 1191 successfully written!
Article 1192 successfully written!
Article 1193 successfully written!
Article 1194 successfully written!
Article 1195 successfully written!
Article 1196 successfully written!
Article 1197 successfully written!
Article 1198 successfully written!
Article 1199 successfully written!
Directory 'page_24.html' created
Article 1200 successfully written!
Article 1201 successfully written!
Article 1202 successfully written!
Article 1203 successfully written!
Article 1204 successfully written!
Article 1205 successfully written!
Article 1206 successfully written!
Article 1207 successfully written!
Article 1208 successfully written!
Article 1209 successfully written!
Article 1210 successfully written!
Article 1211 successfully written!
Article 1212 successfu

Article 1416 successfully written!
Article 1417 successfully written!
Article 1418 successfully written!
Article 1419 successfully written!
Article 1420 successfully written!
Article 1421 successfully written!
Article 1422 successfully written!
Article 1423 successfully written!
Article 1424 successfully written!
Article 1425 successfully written!
Article 1426 successfully written!
Article 1427 successfully written!
Article 1428 successfully written!
Article 1429 successfully written!
Article 1430 successfully written!
Article 1431 successfully written!
Article 1432 successfully written!
Article 1433 successfully written!
Article 1434 successfully written!
Article 1435 successfully written!
Article 1436 successfully written!
Article 1437 successfully written!
Article 1438 successfully written!
Article 1439 successfully written!
Article 1440 successfully written!
Article 1441 successfully written!
Article 1442 successfully written!
Article 1443 successfully written!
Article 1444 success

Article 1648 successfully written!
Article 1649 successfully written!
Directory 'page_33.html' created
Article 1650 successfully written!
Article 1651 successfully written!
Article 1652 successfully written!
Article 1653 successfully written!
Article 1654 successfully written!
Article 1655 successfully written!
Article 1656 successfully written!
Article 1657 successfully written!
Article 1658 successfully written!
Article 1659 successfully written!
Article 1660 successfully written!
Article 1661 successfully written!
Article 1662 successfully written!
Article 1663 successfully written!
Article 1664 successfully written!
Article 1665 successfully written!
Article 1666 successfully written!
Article 1667 successfully written!
Article 1668 successfully written!
Article 1669 successfully written!
Article 1670 successfully written!
Article 1671 successfully written!
Article 1672 successfully written!
Article 1673 successfully written!
Article 1674 successfully written!
Article 1675 successfu

Article 1878 successfully written!
Article 1879 successfully written!
Article 1880 successfully written!
Article 1881 successfully written!
Article 1882 successfully written!
Article 1883 successfully written!
Article 1884 successfully written!
Article 1885 successfully written!
Article 1886 successfully written!
Article 1887 successfully written!
Article 1888 successfully written!
Article 1889 successfully written!
Article 1890 successfully written!
Article 1891 successfully written!
Article 1892 successfully written!
Article 1893 successfully written!
Article 1894 successfully written!
Article 1895 successfully written!
Article 1896 successfully written!
Article 1897 successfully written!
Article 1898 successfully written!
Article 1899 successfully written!
Directory 'page_38.html' created
Article 1900 successfully written!
Article 1901 successfully written!
Article 1902 successfully written!
Article 1903 successfully written!
Article 1904 successfully written!
Article 1905 successfu

Article 2108 successfully written!
Article 2109 successfully written!
Article 2110 successfully written!
Article 2111 successfully written!
Article 2112 successfully written!
Article 2113 successfully written!
Article 2114 successfully written!
Article 2115 successfully written!
Article 2116 successfully written!
Article 2117 successfully written!
Article 2118 successfully written!
Article 2119 successfully written!
Article 2120 successfully written!
Article 2121 successfully written!
Article 2122 successfully written!
Article 2123 successfully written!
Article 2124 successfully written!
Article 2125 successfully written!
Article 2126 successfully written!
Article 2127 successfully written!
Article 2128 successfully written!
Article 2129 successfully written!
Article 2130 successfully written!
Article 2131 successfully written!
Article 2132 successfully written!
Article 2133 successfully written!
Article 2134 successfully written!
Article 2135 successfully written!
Article 2136 success

Article 2340 successfully written!
Article 2341 successfully written!
Article 2342 successfully written!
Article 2343 successfully written!
Article 2344 successfully written!
Article 2345 successfully written!
Article 2346 successfully written!
Article 2347 successfully written!
Article 2348 successfully written!
Article 2349 successfully written!
Directory 'page_47.html' created
Article 2350 successfully written!
Article 2351 successfully written!
Article 2352 successfully written!
Article 2353 successfully written!
Article 2354 successfully written!
Article 2355 successfully written!
Article 2356 successfully written!
Article 2357 successfully written!
Article 2358 successfully written!
Article 2359 successfully written!
Article 2360 successfully written!
Article 2361 successfully written!
Article 2362 successfully written!
Article 2363 successfully written!
Article 2364 successfully written!
Article 2365 successfully written!
Article 2366 successfully written!
Article 2367 successfu

Article 2570 successfully written!
Article 2571 successfully written!
Article 2572 successfully written!
Article 2573 successfully written!
Article 2574 successfully written!
Article 2575 successfully written!
Article 2576 successfully written!
Article 2577 successfully written!
Article 2578 successfully written!
Article 2579 successfully written!
Article 2580 successfully written!
Article 2581 successfully written!
Article 2582 successfully written!
Article 2583 successfully written!
Article 2584 successfully written!
Article 2585 successfully written!
Article 2586 successfully written!
Article 2587 successfully written!
Article 2588 successfully written!
Article 2589 successfully written!
Article 2590 successfully written!
Article 2591 successfully written!
Article 2592 successfully written!
Article 2593 successfully written!
Article 2594 successfully written!
Article 2595 successfully written!
Article 2596 successfully written!
Article 2597 successfully written!
Article 2598 success

Article 2800 successfully written!
Article 2801 successfully written!
Article 2802 successfully written!
Article 2803 successfully written!
Article 2804 successfully written!
Article 2805 successfully written!
Article 2806 successfully written!
Article 2807 successfully written!
Article 2808 successfully written!
Article 2809 successfully written!
Article 2810 successfully written!
Article 2811 successfully written!
Article 2812 successfully written!
Article 2813 successfully written!
Article 2814 successfully written!
Article 2815 successfully written!
Article 2816 successfully written!
Article 2817 successfully written!
Article 2818 successfully written!
Article 2819 successfully written!
Article 2820 successfully written!
Article 2821 successfully written!
Article 2822 successfully written!
Article 2823 successfully written!
Article 2824 successfully written!
Article 2825 successfully written!
Article 2826 successfully written!
Article 2827 successfully written!
Article 2828 success

Article 3032 successfully written!
Article 3033 successfully written!
Article 3034 successfully written!
Article 3035 successfully written!
Article 3036 successfully written!
Article 3037 successfully written!
Article 3038 successfully written!
Article 3039 successfully written!
Article 3040 successfully written!
Article 3041 successfully written!
Article 3042 successfully written!
Article 3043 successfully written!
Article 3044 successfully written!
Article 3045 successfully written!
Article 3046 successfully written!
Article 3047 successfully written!
Article 3048 successfully written!
Article 3049 successfully written!
Directory 'page_61.html' created
Article 3050 successfully written!
Article 3051 successfully written!
Article 3052 successfully written!
Article 3053 successfully written!
Article 3054 successfully written!
Article 3055 successfully written!
Article 3056 successfully written!
Article 3057 successfully written!
Article 3058 successfully written!
Article 3059 successfu

Article 3262 successfully written!
Article 3263 successfully written!
Article 3264 successfully written!
Article 3265 successfully written!
Article 3266 successfully written!
Article 3267 successfully written!
Article 3268 successfully written!
Article 3269 successfully written!
Article 3270 successfully written!
Article 3271 successfully written!
Article 3272 successfully written!
Article 3273 successfully written!
Article 3274 successfully written!
Article 3275 successfully written!
Article 3276 successfully written!
Article 3277 successfully written!
Article 3278 successfully written!
Article 3279 successfully written!
Article 3280 successfully written!
Article 3281 successfully written!
Article 3282 successfully written!
Article 3283 successfully written!
Article 3284 successfully written!
Article 3285 successfully written!
Article 3286 successfully written!
Article 3287 successfully written!
Article 3288 successfully written!
Article 3289 successfully written!
Article 3290 success

Article 3494 successfully written!
Article 3495 successfully written!
Article 3496 successfully written!
Article 3497 successfully written!
Article 3498 successfully written!
Article 3499 successfully written!
Directory 'page_70.html' created
Article 3500 successfully written!
Article 3501 successfully written!
Article 3502 successfully written!
Article 3503 successfully written!
Article 3504 successfully written!
Article 3505 successfully written!
Article 3506 successfully written!
Article 3507 successfully written!
Article 3508 successfully written!
Article 3509 successfully written!
Article 3510 successfully written!
Article 3511 successfully written!
Article 3512 successfully written!
Article 3513 successfully written!
Article 3514 successfully written!
Article 3515 successfully written!
Article 3516 successfully written!
Article 3517 successfully written!
Article 3518 successfully written!
Article 3519 successfully written!
Article 3520 successfully written!
Article 3521 successfu

Article 3725 successfully written!
Article 3726 successfully written!
Article 3727 successfully written!
Article 3728 successfully written!
Article 3729 successfully written!
Article 3730 successfully written!
Article 3731 successfully written!
Article 3732 successfully written!
Article 3733 successfully written!
Article 3734 successfully written!
Article 3735 successfully written!
Article 3736 successfully written!
Article 3737 successfully written!
Article 3738 successfully written!
Article 3739 successfully written!
Article 3740 successfully written!
Article 3741 successfully written!
Article 3742 successfully written!
Article 3743 successfully written!
Article 3744 successfully written!
Article 3745 successfully written!
Article 3746 successfully written!
Article 3747 successfully written!
Article 3748 successfully written!
Article 3749 successfully written!
Directory 'page_75.html' created
Article 3750 successfully written!
Article 3751 successfully written!
Article 3752 successfu

Article 3956 successfully written!
Article 3957 successfully written!
Article 3958 successfully written!
Article 3959 successfully written!
Article 3960 successfully written!
Article 3961 successfully written!
Article 3962 successfully written!
Article 3963 successfully written!
Article 3964 successfully written!
Article 3965 successfully written!
Article 3966 successfully written!
Article 3967 successfully written!
Article 3968 successfully written!
Article 3969 successfully written!
Article 3970 successfully written!
Article 3971 successfully written!
Article 3972 successfully written!
Article 3973 successfully written!
Article 3974 successfully written!
Article 3975 successfully written!
Article 3976 successfully written!
Article 3977 successfully written!
Article 3978 successfully written!
Article 3979 successfully written!
Article 3980 successfully written!
Article 3981 successfully written!
Article 3982 successfully written!
Article 3983 successfully written!
Article 3984 success

Article 4187 successfully written!
Article 4188 successfully written!
Article 4189 successfully written!
Article 4190 successfully written!
Article 4191 successfully written!
Article 4192 successfully written!
Article 4193 successfully written!
Article 4194 successfully written!
Article 4195 successfully written!
Article 4196 successfully written!
Article 4197 successfully written!
Article 4198 successfully written!
Article 4199 successfully written!
Directory 'page_84.html' created
Article 4200 successfully written!
Article 4201 successfully written!
Article 4202 successfully written!
Article 4203 successfully written!
Article 4204 successfully written!
Article 4205 successfully written!
Article 4206 successfully written!
Article 4207 successfully written!
Article 4208 successfully written!
Article 4209 successfully written!
Article 4210 successfully written!
Article 4211 successfully written!
Article 4212 successfully written!
Article 4213 successfully written!
Article 4214 successfu

Article 4418 successfully written!
Article 4419 successfully written!
Article 4420 successfully written!
Article 4421 successfully written!
Article 4422 successfully written!
Article 4423 successfully written!
Article 4424 successfully written!
Article 4425 successfully written!
Article 4426 successfully written!
Article 4427 successfully written!
Article 4428 successfully written!
Article 4429 successfully written!
Article 4430 successfully written!
Article 4431 successfully written!
Article 4432 successfully written!
Article 4433 successfully written!
Article 4434 successfully written!
Article 4435 successfully written!
Article 4436 successfully written!
Article 4437 successfully written!
Article 4438 successfully written!
Article 4439 successfully written!
Article 4440 successfully written!
Article 4441 successfully written!
Article 4442 successfully written!
Article 4443 successfully written!
Article 4444 successfully written!
Article 4445 successfully written!
Article 4446 success

Article 4649 successfully written!
Directory 'page_93.html' created
Article 4650 successfully written!
Article 4651 successfully written!
Article 4652 successfully written!
Article 4653 successfully written!
Article 4654 successfully written!
Article 4655 successfully written!
Article 4656 successfully written!
Article 4657 successfully written!
Article 4658 successfully written!
Article 4659 successfully written!
Article 4660 successfully written!
Article 4661 successfully written!
Article 4662 successfully written!
Article 4663 successfully written!
Article 4664 successfully written!
Article 4665 successfully written!
Article 4666 successfully written!
Article 4667 successfully written!
Article 4668 successfully written!
Article 4669 successfully written!
Article 4670 successfully written!
Article 4671 successfully written!
Article 4672 successfully written!
Article 4673 successfully written!
Article 4674 successfully written!
Article 4675 successfully written!
Article 4676 successfu

Article 4879 successfully written!
Article 4880 successfully written!
Article 4881 successfully written!
Article 4882 successfully written!
Article 4883 successfully written!
Article 4884 successfully written!
Article 4885 successfully written!
Article 4886 successfully written!
Article 4887 successfully written!
Article 4888 successfully written!
Article 4889 successfully written!
Article 4890 successfully written!
Article 4891 successfully written!
Article 4892 successfully written!
Article 4893 successfully written!
Article 4894 successfully written!
Article 4895 successfully written!
Article 4896 successfully written!
Article 4897 successfully written!
Article 4898 successfully written!
Article 4899 successfully written!
Directory 'page_98.html' created
Article 4900 successfully written!
Article 4901 successfully written!
Article 4902 successfully written!
Article 4903 successfully written!
Article 4904 successfully written!
Article 4905 successfully written!
Article 4906 successfu

Article 5110 successfully written!
Article 5111 successfully written!
Article 5112 successfully written!
Article 5113 successfully written!
Article 5114 successfully written!
Article 5115 successfully written!
Article 5116 successfully written!
Article 5117 successfully written!
Article 5118 successfully written!
Article 5119 successfully written!
Article 5120 successfully written!
Article 5121 successfully written!
Article 5122 successfully written!
Article 5123 successfully written!
Article 5124 successfully written!
Article 5125 successfully written!
Article 5126 successfully written!
Article 5127 successfully written!
Article 5128 successfully written!
Article 5129 successfully written!
Article 5130 successfully written!
Article 5131 successfully written!
Article 5132 successfully written!
Article 5133 successfully written!
Article 5134 successfully written!
Article 5135 successfully written!
Article 5136 successfully written!
Article 5137 successfully written!
Article 5138 success

Article 5342 successfully written!
Article 5343 successfully written!
Article 5344 successfully written!
Article 5345 successfully written!
Article 5346 successfully written!
Article 5347 successfully written!
Article 5348 successfully written!
Article 5349 successfully written!
Directory 'page_107.html' created
Article 5350 successfully written!
Article 5351 successfully written!
Article 5352 successfully written!
Article 5353 successfully written!
Article 5354 successfully written!
Article 5355 successfully written!
Article 5356 successfully written!
Article 5357 successfully written!
Article 5358 successfully written!
Article 5359 successfully written!
Article 5360 successfully written!
Article 5361 successfully written!
Article 5362 successfully written!
Article 5363 successfully written!
Article 5364 successfully written!
Article 5365 successfully written!
Article 5366 successfully written!
Article 5367 successfully written!
Article 5368 successfully written!
Article 5369 successf

Article 5572 successfully written!
Article 5573 successfully written!
Article 5574 successfully written!
Article 5575 successfully written!
Article 5576 successfully written!
Article 5577 successfully written!
Article 5578 successfully written!
Article 5579 successfully written!
Article 5580 successfully written!
Article 5581 successfully written!
Article 5582 successfully written!
Article 5583 successfully written!
Article 5584 successfully written!
Article 5585 successfully written!
Article 5586 successfully written!
Article 5587 successfully written!
Article 5588 successfully written!
Article 5589 successfully written!
Article 5590 successfully written!
Article 5591 successfully written!
Article 5592 successfully written!
Article 5593 successfully written!
Article 5594 successfully written!
Article 5595 successfully written!
Article 5596 successfully written!
Article 5597 successfully written!
Article 5598 successfully written!
Article 5599 successfully written!
Directory 'page_112.

Article 5802 successfully written!
Article 5803 successfully written!
Article 5804 successfully written!
Article 5805 successfully written!
Article 5806 successfully written!
Article 5807 successfully written!
Article 5808 successfully written!
Article 5809 successfully written!
Article 5810 successfully written!
Article 5811 successfully written!
Article 5812 successfully written!
Article 5813 successfully written!
Article 5814 successfully written!
Article 5815 successfully written!
Article 5816 successfully written!
Article 5817 successfully written!
Article 5818 successfully written!
Article 5819 successfully written!
Article 5820 successfully written!
Article 5821 successfully written!
Article 5822 successfully written!
Article 5823 successfully written!
Article 5824 successfully written!
Article 5825 successfully written!
Article 5826 successfully written!
Article 5827 successfully written!
Article 5828 successfully written!
Article 5829 successfully written!
Article 5830 success

Article 6034 successfully written!
Article 6035 successfully written!
Article 6036 successfully written!
Article 6037 successfully written!
Article 6038 successfully written!
Article 6039 successfully written!
Article 6040 successfully written!
Article 6041 successfully written!
Article 6042 successfully written!
Article 6043 successfully written!
Article 6044 successfully written!
Article 6045 successfully written!
Article 6046 successfully written!
Article 6047 successfully written!
Article 6048 successfully written!
Article 6049 successfully written!
Directory 'page_121.html' created
Article 6050 successfully written!
Article 6051 successfully written!
Article 6052 successfully written!
Article 6053 successfully written!
Article 6054 successfully written!
Article 6055 successfully written!
Article 6056 successfully written!
Article 6057 successfully written!
Article 6058 successfully written!
Article 6059 successfully written!
Article 6060 successfully written!
Article 6061 successf

Article 6265 successfully written!
Article 6266 successfully written!
Article 6267 successfully written!
Article 6268 successfully written!
Article 6269 successfully written!
Article 6270 successfully written!
Article 6271 successfully written!
Article 6272 successfully written!
Article 6273 successfully written!
Article 6274 successfully written!
Article 6275 successfully written!
Article 6276 successfully written!
Article 6277 successfully written!
Article 6278 successfully written!
Article 6279 successfully written!
Article 6280 successfully written!
Article 6281 successfully written!
Article 6282 successfully written!
Article 6283 successfully written!
Article 6284 successfully written!
Article 6285 successfully written!
Article 6286 successfully written!
Article 6287 successfully written!
Article 6288 successfully written!
Article 6289 successfully written!
Article 6290 successfully written!
Article 6291 successfully written!
Article 6292 successfully written!
Article 6293 success

Article 6497 successfully written!
Article 6498 successfully written!
Article 6499 successfully written!
Directory 'page_130.html' created
Article 6500 successfully written!
Article 6501 successfully written!
Article 6502 successfully written!
Article 6503 successfully written!
Article 6504 successfully written!
Article 6505 successfully written!
Article 6506 successfully written!
Article 6507 successfully written!
Article 6508 successfully written!
Article 6509 successfully written!
Article 6510 successfully written!
Article 6511 successfully written!
Article 6512 successfully written!
Article 6513 successfully written!
Article 6514 successfully written!
Article 6515 successfully written!
Article 6516 successfully written!
Article 6517 successfully written!
Article 6518 successfully written!
Article 6519 successfully written!
Article 6520 successfully written!
Article 6521 successfully written!
Article 6522 successfully written!
Article 6523 successfully written!
Article 6524 successf

Article 6728 successfully written!
Article 6729 successfully written!
Article 6730 successfully written!
Article 6731 successfully written!
Article 6732 successfully written!
Article 6733 successfully written!
Article 6734 successfully written!
Article 6735 successfully written!
Article 6736 successfully written!
Article 6737 successfully written!
Article 6738 successfully written!
Article 6739 successfully written!
Article 6740 successfully written!
Article 6741 successfully written!
Article 6742 successfully written!
Article 6743 successfully written!
Article 6744 successfully written!
Article 6745 successfully written!
Article 6746 successfully written!
Article 6747 successfully written!
Article 6748 successfully written!
Article 6749 successfully written!
Directory 'page_135.html' created
Article 6750 successfully written!
Article 6751 successfully written!
Article 6752 successfully written!
Article 6753 successfully written!
Article 6754 successfully written!
Article 6755 successf

Article 6959 successfully written!
Article 6960 successfully written!
Article 6961 successfully written!
Article 6962 successfully written!
Article 6963 successfully written!
Article 6964 successfully written!
Article 6965 successfully written!
Article 6966 successfully written!
Article 6967 successfully written!
Article 6968 successfully written!
Article 6969 successfully written!
Article 6970 successfully written!
Article 6971 successfully written!
Article 6972 successfully written!
Article 6973 successfully written!
Article 6974 successfully written!
Article 6975 successfully written!
Article 6976 successfully written!
Article 6977 successfully written!
Article 6978 successfully written!
Article 6979 successfully written!
Article 6980 successfully written!
Article 6981 successfully written!
Article 6982 successfully written!
Article 6983 successfully written!
Article 6984 successfully written!
Article 6985 successfully written!
Article 6986 successfully written!
Article 6987 success

Article 7190 successfully written!
Article 7191 successfully written!
Article 7192 successfully written!
Article 7193 successfully written!
Article 7194 successfully written!
Article 7195 successfully written!
Article 7196 successfully written!
Article 7197 successfully written!
Article 7198 successfully written!
Article 7199 successfully written!
Directory 'page_144.html' created
Article 7200 successfully written!
Article 7201 successfully written!
Article 7202 successfully written!
Article 7203 successfully written!
Article 7204 successfully written!
Article 7205 successfully written!
Article 7206 successfully written!
Article 7207 successfully written!
Article 7208 successfully written!
Article 7209 successfully written!
Article 7210 successfully written!
Article 7211 successfully written!
Article 7212 successfully written!
Article 7213 successfully written!
Article 7214 successfully written!
Article 7215 successfully written!
Article 7216 successfully written!
Article 7217 successf

Article 7420 successfully written!
Article 7421 successfully written!
Article 7422 successfully written!
Article 7423 successfully written!
Article 7424 successfully written!
Article 7425 successfully written!
Article 7426 successfully written!
Article 7427 successfully written!
Article 7428 successfully written!
Article 7429 successfully written!
Article 7430 successfully written!
Article 7431 successfully written!
Article 7432 successfully written!
Article 7433 successfully written!
Article 7434 successfully written!
Article 7435 successfully written!
Article 7436 successfully written!
Article 7437 successfully written!
Article 7438 successfully written!
Article 7439 successfully written!
Article 7440 successfully written!
Article 7441 successfully written!
Article 7442 successfully written!
Article 7443 successfully written!
Article 7444 successfully written!
Article 7445 successfully written!
Article 7446 successfully written!
Article 7447 successfully written!
Article 7448 success

Article 7650 successfully written!
Article 7651 successfully written!
Article 7652 successfully written!
Article 7653 successfully written!
Article 7654 successfully written!
Article 7655 successfully written!
Article 7656 successfully written!
Article 7657 successfully written!
Article 7658 successfully written!
Article 7659 successfully written!
Article 7660 successfully written!
Article 7661 successfully written!
Article 7662 successfully written!
Article 7663 successfully written!
Article 7664 successfully written!
Article 7665 successfully written!
Article 7666 successfully written!
Article 7667 successfully written!
Article 7668 successfully written!
Article 7669 successfully written!
Article 7670 successfully written!
Article 7671 successfully written!
Article 7672 successfully written!
Article 7673 successfully written!
Article 7674 successfully written!
Article 7675 successfully written!
Article 7676 successfully written!
Article 7677 successfully written!
Article 7678 success

Article 7881 successfully written!
Article 7882 successfully written!
Article 7883 successfully written!
Article 7884 successfully written!
Article 7885 successfully written!
Article 7886 successfully written!
Article 7887 successfully written!
Article 7888 successfully written!
Article 7889 successfully written!
Article 7890 successfully written!
Article 7891 successfully written!
Article 7892 successfully written!
Article 7893 successfully written!
Article 7894 successfully written!
Article 7895 successfully written!
Article 7896 successfully written!
Article 7897 successfully written!
Article 7898 successfully written!
Article 7899 successfully written!
Directory 'page_158.html' created
Article 7900 successfully written!
Article 7901 successfully written!
Article 7902 successfully written!
Article 7903 successfully written!
Article 7904 successfully written!
Article 7905 successfully written!
Article 7906 successfully written!
Article 7907 successfully written!
Article 7908 successf

Article 8111 successfully written!
Article 8112 successfully written!
Article 8113 successfully written!
Article 8114 successfully written!
Article 8115 successfully written!
Article 8116 successfully written!
Article 8117 successfully written!
Article 8118 successfully written!
Article 8119 successfully written!
Article 8120 successfully written!
Article 8121 successfully written!
Article 8122 successfully written!
Article 8123 successfully written!
Article 8124 successfully written!
Article 8125 successfully written!
Article 8126 successfully written!
Article 8127 successfully written!
Article 8128 successfully written!
Article 8129 successfully written!
Article 8130 successfully written!
Article 8131 successfully written!
Article 8132 successfully written!
Article 8133 successfully written!
Article 8134 successfully written!
Article 8135 successfully written!
Article 8136 successfully written!
Article 8137 successfully written!
Article 8138 successfully written!
Article 8139 success

Article 8343 successfully written!
Article 8344 successfully written!
Article 8345 successfully written!
Article 8346 successfully written!
Article 8347 successfully written!
Article 8348 successfully written!
Article 8349 successfully written!
Directory 'page_167.html' created
Article 8350 successfully written!
Article 8351 successfully written!
Article 8352 successfully written!
Article 8353 successfully written!
Article 8354 successfully written!
Article 8355 successfully written!
Article 8356 successfully written!
Article 8357 successfully written!
Article 8358 successfully written!
Article 8359 successfully written!
Article 8360 successfully written!
Article 8361 successfully written!
Article 8362 successfully written!
Article 8363 successfully written!
Article 8364 successfully written!
Article 8365 successfully written!
Article 8366 successfully written!
Article 8367 successfully written!
Article 8368 successfully written!
Article 8369 successfully written!
Article 8370 successf

Article 8573 successfully written!
Article 8574 successfully written!
Article 8575 successfully written!
Article 8576 successfully written!
Article 8577 successfully written!
Article 8578 successfully written!
Article 8579 successfully written!
Article 8580 successfully written!
Article 8581 successfully written!
Article 8582 successfully written!
Article 8583 successfully written!
Article 8584 successfully written!
Article 8585 successfully written!
Article 8586 successfully written!
Article 8587 successfully written!
Article 8588 successfully written!
Article 8589 successfully written!
Article 8590 successfully written!
Article 8591 successfully written!
Article 8592 successfully written!
Article 8593 successfully written!
Article 8594 successfully written!
Article 8595 successfully written!
Article 8596 successfully written!
Article 8597 successfully written!
Article 8598 successfully written!
Article 8599 successfully written!
Directory 'page_172.html' created
Article 8600 successf

Article 8803 successfully written!
Article 8804 successfully written!
Article 8805 successfully written!
Article 8806 successfully written!
Article 8807 successfully written!
Article 8808 successfully written!
Article 8809 successfully written!
Article 8810 successfully written!
Article 8811 successfully written!
Article 8812 successfully written!
Article 8813 successfully written!
Article 8814 successfully written!
Article 8815 successfully written!
Article 8816 successfully written!
Article 8817 successfully written!
Article 8818 successfully written!
Article 8819 successfully written!
Article 8820 successfully written!
Article 8821 successfully written!
Article 8822 successfully written!
Article 8823 successfully written!
Article 8824 successfully written!
Article 8825 successfully written!
Article 8826 successfully written!
Article 8827 successfully written!
Article 8828 successfully written!
Article 8829 successfully written!
Article 8830 successfully written!
Article 8831 success

Article 9035 successfully written!
Article 9036 successfully written!
Article 9037 successfully written!
Article 9038 successfully written!
Article 9039 successfully written!
Article 9040 successfully written!
Article 9041 successfully written!
Article 9042 successfully written!
Article 9043 successfully written!
Article 9044 successfully written!
Article 9045 successfully written!
Article 9046 successfully written!
Article 9047 successfully written!
Article 9048 successfully written!
Article 9049 successfully written!
Directory 'page_181.html' created
Article 9050 successfully written!
Article 9051 successfully written!
Article 9052 successfully written!
Article 9053 successfully written!
Article 9054 successfully written!
Article 9055 successfully written!
Article 9056 successfully written!
Article 9057 successfully written!
Article 9058 successfully written!
Article 9059 successfully written!
Article 9060 successfully written!
Article 9061 successfully written!
Article 9062 successf

Article 9266 successfully written!
Article 9267 successfully written!
Article 9268 successfully written!
Article 9269 successfully written!
Article 9270 successfully written!
Article 9271 successfully written!
Article 9272 successfully written!
Article 9273 successfully written!
Article 9274 successfully written!
Article 9275 successfully written!
Article 9276 successfully written!
Article 9277 successfully written!
Article 9278 successfully written!
Article 9279 successfully written!
Article 9280 successfully written!
Article 9281 successfully written!
Article 9282 successfully written!
Article 9283 successfully written!
Article 9284 successfully written!
Article 9285 successfully written!
Article 9286 successfully written!
Article 9287 successfully written!
Article 9288 successfully written!
Article 9289 successfully written!
Article 9290 successfully written!
Article 9291 successfully written!
Article 9292 successfully written!
Article 9293 successfully written!
Article 9294 success

Article 9497 successfully written!
Article 9498 successfully written!
Article 9499 successfully written!
Directory 'page_190.html' created
Article 9500 successfully written!
Article 9501 successfully written!
Article 9502 successfully written!
Article 9503 successfully written!
Article 9504 successfully written!
Article 9505 successfully written!
Article 9506 successfully written!
Article 9507 successfully written!
Article 9508 successfully written!
Article 9509 successfully written!
Article 9510 successfully written!
Article 9511 successfully written!
Article 9512 successfully written!
Article 9513 successfully written!
Article 9514 successfully written!
Article 9515 successfully written!
Article 9516 successfully written!
Article 9517 successfully written!
Article 9518 successfully written!
Article 9519 successfully written!
Article 9520 successfully written!
Article 9521 successfully written!
Article 9522 successfully written!
Article 9523 successfully written!
Article 9524 successf

Article 9727 successfully written!
Article 9728 successfully written!
Article 9729 successfully written!
Article 9730 successfully written!
Article 9731 successfully written!
Article 9732 successfully written!
Article 9733 successfully written!
Article 9734 successfully written!
Article 9735 successfully written!
Article 9736 successfully written!
Article 9737 successfully written!
Article 9738 successfully written!
Article 9739 successfully written!
Article 9740 successfully written!
Article 9741 successfully written!
Article 9742 successfully written!
Article 9743 successfully written!
Article 9744 successfully written!
Article 9745 successfully written!
Article 9746 successfully written!
Article 9747 successfully written!
Article 9748 successfully written!
Article 9749 successfully written!
Directory 'page_195.html' created
Article 9750 successfully written!
Article 9751 successfully written!
Article 9752 successfully written!
Article 9753 successfully written!
Article 9754 successf

Article 9957 successfully written!
Article 9958 successfully written!
Article 9959 successfully written!
Article 9960 successfully written!
Article 9961 successfully written!
Article 9962 successfully written!
Article 9963 successfully written!
Article 9964 successfully written!
Article 9965 successfully written!
Article 9966 successfully written!
Article 9967 successfully written!
Article 9968 successfully written!
Article 9969 successfully written!
Article 9970 successfully written!
Article 9971 successfully written!
Article 9972 successfully written!
Article 9973 successfully written!
Article 9974 successfully written!
Article 9975 successfully written!
Article 9976 successfully written!
Article 9977 successfully written!
Article 9978 successfully written!
Article 9979 successfully written!
Article 9980 successfully written!
Article 9981 successfully written!
Article 9982 successfully written!
Article 9983 successfully written!
Article 9984 successfully written!
Article 9985 success

Article 10183 successfully written!
Article 10184 successfully written!
Article 10185 successfully written!
Article 10186 successfully written!
Article 10187 successfully written!
Article 10188 successfully written!
Article 10189 successfully written!
Article 10190 successfully written!
Article 10191 successfully written!
Article 10192 successfully written!
Article 10193 successfully written!
Article 10194 successfully written!
Article 10195 successfully written!
Article 10196 successfully written!
Article 10197 successfully written!
Article 10198 successfully written!
Article 10199 successfully written!
Directory 'page_204.html' created
Article 10200 successfully written!
Article 10201 successfully written!
Article 10202 successfully written!
Article 10203 successfully written!
Article 10204 successfully written!
Article 10205 successfully written!
Article 10206 successfully written!
Article 10207 successfully written!
Article 10208 successfully written!
Article 10209 successfully wri

Article 10407 successfully written!
Article 10408 successfully written!
Article 10409 successfully written!
Article 10410 successfully written!
Article 10411 successfully written!
Article 10412 successfully written!
Article 10413 successfully written!
Article 10414 successfully written!
Article 10415 successfully written!
Article 10416 successfully written!
Article 10417 successfully written!
Article 10418 successfully written!
Article 10419 successfully written!
Article 10420 successfully written!
Article 10421 successfully written!
Article 10422 successfully written!
Article 10423 successfully written!
Article 10424 successfully written!
Article 10425 successfully written!
Article 10426 successfully written!
Article 10427 successfully written!
Article 10428 successfully written!
Article 10429 successfully written!
Article 10430 successfully written!
Article 10431 successfully written!
Article 10432 successfully written!
Article 10433 successfully written!
Article 10434 successfully w

Article 10631 successfully written!
Article 10632 successfully written!
Article 10633 successfully written!
Article 10634 successfully written!
Article 10635 successfully written!
Article 10636 successfully written!
Article 10637 successfully written!
Article 10638 successfully written!
Article 10639 successfully written!
Article 10640 successfully written!
Article 10641 successfully written!
Article 10642 successfully written!
Article 10643 successfully written!
Article 10644 successfully written!
Article 10645 successfully written!
Article 10646 successfully written!
Article 10647 successfully written!
Article 10648 successfully written!
Article 10649 successfully written!
Directory 'page_213.html' created
Article 10650 successfully written!
Article 10651 successfully written!
Article 10652 successfully written!
Article 10653 successfully written!
Article 10654 successfully written!
Article 10655 successfully written!
Article 10656 successfully written!
Article 10657 successfully wri

Article 10855 successfully written!
Article 10856 successfully written!
Article 10857 successfully written!
Article 10858 successfully written!
Article 10859 successfully written!
Article 10860 successfully written!
Article 10861 successfully written!
Article 10862 successfully written!
Article 10863 successfully written!
Article 10864 successfully written!
Article 10865 successfully written!
Article 10866 successfully written!
Article 10867 successfully written!
Article 10868 successfully written!
Article 10869 successfully written!
Article 10870 successfully written!
Article 10871 successfully written!
Article 10872 successfully written!
Article 10873 successfully written!
Article 10874 successfully written!
Article 10875 successfully written!
Article 10876 successfully written!
Article 10877 successfully written!
Article 10878 successfully written!
Article 10879 successfully written!
Article 10880 successfully written!
Article 10881 successfully written!
Article 10882 successfully w

Article 11080 successfully written!
Article 11081 successfully written!
Article 11082 successfully written!
Article 11083 successfully written!
Article 11084 successfully written!
Article 11085 successfully written!
Article 11086 successfully written!
Article 11087 successfully written!
Article 11088 successfully written!
Article 11089 successfully written!
Article 11090 successfully written!
Article 11091 successfully written!
Article 11092 successfully written!
Article 11093 successfully written!
Article 11094 successfully written!
Article 11095 successfully written!
Article 11096 successfully written!
Article 11097 successfully written!
Article 11098 successfully written!
Article 11099 successfully written!
Directory 'page_222.html' created
Article 11100 successfully written!
Article 11101 successfully written!
Article 11102 successfully written!
Article 11103 successfully written!
Article 11104 successfully written!
Article 11105 successfully written!
Article 11106 successfully wri

Article 11304 successfully written!
Article 11305 successfully written!
Article 11306 successfully written!
Article 11307 successfully written!
Article 11308 successfully written!
Article 11309 successfully written!
Article 11310 successfully written!
Article 11311 successfully written!
Article 11312 successfully written!
Article 11313 successfully written!
Article 11314 successfully written!
Article 11315 successfully written!
Article 11316 successfully written!
Article 11317 successfully written!
Article 11318 successfully written!
Article 11319 successfully written!
Article 11320 successfully written!
Article 11321 successfully written!
Article 11322 successfully written!
Article 11323 successfully written!
Article 11324 successfully written!
Article 11325 successfully written!
Article 11326 successfully written!
Article 11327 successfully written!
Article 11328 successfully written!
Article 11329 successfully written!
Article 11330 successfully written!
Article 11331 successfully w

Article 11528 successfully written!
Article 11529 successfully written!
Article 11530 successfully written!
Article 11531 successfully written!
Article 11532 successfully written!
Article 11533 successfully written!
Article 11534 successfully written!
Article 11535 successfully written!
Article 11536 successfully written!
Article 11537 successfully written!
Article 11538 successfully written!
Article 11539 successfully written!
Article 11540 successfully written!
Article 11541 successfully written!
Article 11542 successfully written!
Article 11543 successfully written!
Article 11544 successfully written!
Article 11545 successfully written!
Article 11546 successfully written!
Article 11547 successfully written!
Article 11548 successfully written!
Article 11549 successfully written!
Directory 'page_231.html' created
Article 11550 successfully written!
Article 11551 successfully written!
Article 11552 successfully written!
Article 11553 successfully written!
Article 11554 successfully wri

Article 11752 successfully written!
Article 11753 successfully written!
Article 11754 successfully written!
Article 11755 successfully written!
Article 11756 successfully written!
Article 11757 successfully written!
Article 11758 successfully written!
Article 11759 successfully written!
Article 11760 successfully written!
Article 11761 successfully written!
Article 11762 successfully written!
Article 11763 successfully written!
Article 11764 successfully written!
Article 11765 successfully written!
Article 11766 successfully written!
Article 11767 successfully written!
Article 11768 successfully written!
Article 11769 successfully written!
Article 11770 successfully written!
Article 11771 successfully written!
Article 11772 successfully written!
Article 11773 successfully written!
Article 11774 successfully written!
Article 11775 successfully written!
Article 11776 successfully written!
Article 11777 successfully written!
Article 11778 successfully written!
Article 11779 successfully w

Article 11976 successfully written!
Article 11977 successfully written!
Article 11978 successfully written!
Article 11979 successfully written!
Article 11980 successfully written!
Article 11981 successfully written!
Article 11982 successfully written!
Article 11983 successfully written!
Article 11984 successfully written!
Article 11985 successfully written!
Article 11986 successfully written!
Article 11987 successfully written!
Article 11988 successfully written!
Article 11989 successfully written!
Article 11990 successfully written!
Article 11991 successfully written!
Article 11992 successfully written!
Article 11993 successfully written!
Article 11994 successfully written!
Article 11995 successfully written!
Article 11996 successfully written!
Article 11997 successfully written!
Article 11998 successfully written!
Article 11999 successfully written!
Directory 'page_240.html' created
Article 12000 successfully written!
Article 12001 successfully written!
Article 12002 successfully wri

Article 12200 successfully written!
Article 12201 successfully written!
Article 12202 successfully written!
Article 12203 successfully written!
Article 12204 successfully written!
Article 12205 successfully written!
Article 12206 successfully written!
Article 12207 successfully written!
Article 12208 successfully written!
Article 12209 successfully written!
Article 12210 successfully written!
Article 12211 successfully written!
Article 12212 successfully written!
Article 12213 successfully written!
Article 12214 successfully written!
Article 12215 successfully written!
Article 12216 successfully written!
Article 12217 successfully written!
Article 12218 successfully written!
Article 12219 successfully written!
Article 12220 successfully written!
Article 12221 successfully written!
Article 12222 successfully written!
Article 12223 successfully written!
Article 12224 successfully written!
Article 12225 successfully written!
Article 12226 successfully written!
Article 12227 successfully w

Article 12425 successfully written!
Article 12426 successfully written!
Article 12427 successfully written!
Article 12428 successfully written!
Article 12429 successfully written!
Article 12430 successfully written!
Article 12431 successfully written!
Article 12432 successfully written!
Article 12433 successfully written!
Article 12434 successfully written!
Article 12435 successfully written!
Article 12436 successfully written!
Article 12437 successfully written!
Article 12438 successfully written!
Article 12439 successfully written!
Article 12440 successfully written!
Article 12441 successfully written!
Article 12442 successfully written!
Article 12443 successfully written!
Article 12444 successfully written!
Article 12445 successfully written!
Article 12446 successfully written!
Article 12447 successfully written!
Article 12448 successfully written!
Article 12449 successfully written!
Directory 'page_249.html' created
Article 12450 successfully written!
Article 12451 successfully wri

Article 12650 successfully written!
Article 12651 successfully written!
Article 12652 successfully written!
Article 12653 successfully written!
Article 12654 successfully written!
Article 12655 successfully written!
Article 12656 successfully written!
Article 12657 successfully written!
Article 12658 successfully written!
Article 12659 successfully written!
Article 12660 successfully written!
Article 12661 successfully written!
Article 12662 successfully written!
Article 12663 successfully written!
Article 12664 successfully written!
Article 12665 successfully written!
Article 12666 successfully written!
Article 12667 successfully written!
Article 12668 successfully written!
Article 12669 successfully written!
Article 12670 successfully written!
Article 12671 successfully written!
Article 12672 successfully written!
Article 12673 successfully written!
Article 12674 successfully written!
Article 12675 successfully written!
Article 12676 successfully written!
Article 12677 successfully w

Article 12874 successfully written!
Article 12875 successfully written!
Article 12876 successfully written!
Article 12877 successfully written!
Article 12878 successfully written!
Article 12879 successfully written!
Article 12880 successfully written!
Article 12881 successfully written!
Article 12882 successfully written!
Article 12883 successfully written!
Article 12884 successfully written!
Article 12885 successfully written!
Article 12886 successfully written!
Article 12887 successfully written!
Article 12888 successfully written!
Article 12889 successfully written!
Article 12890 successfully written!
Article 12891 successfully written!
Article 12892 successfully written!
Article 12893 successfully written!
Article 12894 successfully written!
Article 12895 successfully written!
Article 12896 successfully written!
Article 12897 successfully written!
Article 12898 successfully written!
Article 12899 successfully written!
Directory 'page_258.html' created
Article 12900 successfully wri

Article 13098 successfully written!
Article 13099 successfully written!
Directory 'page_262.html' created
Article 13100 successfully written!
Article 13101 successfully written!
Article 13102 successfully written!
Article 13103 successfully written!
Article 13104 successfully written!
Article 13105 successfully written!
Article 13106 successfully written!
Article 13107 successfully written!
Article 13108 successfully written!
Article 13109 successfully written!
Article 13110 successfully written!
Article 13111 successfully written!
Article 13112 successfully written!
Article 13113 successfully written!
Article 13114 successfully written!
Article 13115 successfully written!
Article 13116 successfully written!
Article 13117 successfully written!
Article 13118 successfully written!
Article 13119 successfully written!
Article 13120 successfully written!
Article 13121 successfully written!
Article 13122 successfully written!
Article 13123 successfully written!
Article 13124 successfully wri

Article 13322 successfully written!
Article 13323 successfully written!
Article 13324 successfully written!
Article 13325 successfully written!
Article 13326 successfully written!
Article 13327 successfully written!
Article 13328 successfully written!
Article 13329 successfully written!
Article 13330 successfully written!
Article 13331 successfully written!
Article 13332 successfully written!
Article 13333 successfully written!
Article 13334 successfully written!
Article 13335 successfully written!
Article 13336 successfully written!
Article 13337 successfully written!
Article 13338 successfully written!
Article 13339 successfully written!
Article 13340 successfully written!
Article 13341 successfully written!
Article 13342 successfully written!
Article 13343 successfully written!
Article 13344 successfully written!
Article 13345 successfully written!
Article 13346 successfully written!
Article 13347 successfully written!
Article 13348 successfully written!
Article 13349 successfully w

Article 13546 successfully written!
Article 13547 successfully written!
Article 13548 successfully written!
Article 13549 successfully written!
Directory 'page_271.html' created
Article 13550 successfully written!
Article 13551 successfully written!
Article 13552 successfully written!
Article 13553 successfully written!
Article 13554 successfully written!
Article 13555 successfully written!
Article 13556 successfully written!
Article 13557 successfully written!
Article 13558 successfully written!
Article 13559 successfully written!
Article 13560 successfully written!
Article 13561 successfully written!
Article 13562 successfully written!
Article 13563 successfully written!
Article 13564 successfully written!
Article 13565 successfully written!
Article 13566 successfully written!
Article 13567 successfully written!
Article 13568 successfully written!
Article 13569 successfully written!
Article 13570 successfully written!
Article 13571 successfully written!
Article 13572 successfully wri

Article 13770 successfully written!
Article 13771 successfully written!
Article 13772 successfully written!
Article 13773 successfully written!
Article 13774 successfully written!
Article 13775 successfully written!
Article 13776 successfully written!
Article 13777 successfully written!
Article 13778 successfully written!
Article 13779 successfully written!
Article 13780 successfully written!
Article 13781 successfully written!
Article 13782 successfully written!
Article 13783 successfully written!
Article 13784 successfully written!
Article 13785 successfully written!
Article 13786 successfully written!
Article 13787 successfully written!
Article 13788 successfully written!
Article 13789 successfully written!
Article 13790 successfully written!
Article 13791 successfully written!
Article 13792 successfully written!
Article 13793 successfully written!
Article 13794 successfully written!
Article 13795 successfully written!
Article 13796 successfully written!
Article 13797 successfully w

Article 13994 successfully written!
Article 13995 successfully written!
Article 13996 successfully written!
Article 13997 successfully written!
Article 13998 successfully written!
Article 13999 successfully written!
Directory 'page_280.html' created
Article 14000 successfully written!
Article 14001 successfully written!
Article 14002 successfully written!
Article 14003 successfully written!
Article 14004 successfully written!
Article 14005 successfully written!
Article 14006 successfully written!
Article 14007 successfully written!
Article 14008 successfully written!
Article 14009 successfully written!
Article 14010 successfully written!
Article 14011 successfully written!
Article 14012 successfully written!
Article 14013 successfully written!
Article 14014 successfully written!
Article 14015 successfully written!
Article 14016 successfully written!
Article 14017 successfully written!
Article 14018 successfully written!
Article 14019 successfully written!
Article 14020 successfully wri

Article 14217 successfully written!
Article 14218 successfully written!
Article 14219 successfully written!
Article 14220 successfully written!
Article 14221 successfully written!
Article 14222 successfully written!
Article 14223 successfully written!
Article 14224 successfully written!
Article 14225 successfully written!
Article 14226 successfully written!
Article 14227 successfully written!
Article 14228 successfully written!
Article 14229 successfully written!
Article 14230 successfully written!
Article 14231 successfully written!
Article 14232 successfully written!
Article 14233 successfully written!
Article 14234 successfully written!
Article 14235 successfully written!
Article 14236 successfully written!
Article 14237 successfully written!
Article 14238 successfully written!
Article 14239 successfully written!
Article 14240 successfully written!
Article 14241 successfully written!
Article 14242 successfully written!
Article 14243 successfully written!
Article 14244 successfully w

Article 14441 successfully written!
Article 14442 successfully written!
Article 14443 successfully written!
Article 14444 successfully written!
Article 14445 successfully written!
Article 14446 successfully written!
Article 14447 successfully written!
Article 14448 successfully written!
Article 14449 successfully written!
Directory 'page_289.html' created
Article 14450 successfully written!
Article 14451 successfully written!
Article 14452 successfully written!
Article 14453 successfully written!
Article 14454 successfully written!
Article 14455 successfully written!
Article 14456 successfully written!
Article 14457 successfully written!
Article 14458 successfully written!
Article 14459 successfully written!
Article 14460 successfully written!
Article 14461 successfully written!
Article 14462 successfully written!
Article 14463 successfully written!
Article 14464 successfully written!
Article 14465 successfully written!
Article 14466 successfully written!
Article 14467 successfully wri

Article 14664 successfully written!
Article 14665 successfully written!
Article 14666 successfully written!
Article 14667 successfully written!
Article 14668 successfully written!
Article 14669 successfully written!
Article 14670 successfully written!
Article 14671 successfully written!
Article 14672 successfully written!
Article 14673 successfully written!
Article 14674 successfully written!
Article 14675 successfully written!
Article 14676 successfully written!
Article 14677 successfully written!
Article 14678 successfully written!
Article 14679 successfully written!
Article 14680 successfully written!
Article 14681 successfully written!
Article 14682 successfully written!
Article 14683 successfully written!
Article 14684 successfully written!
Article 14685 successfully written!
Article 14686 successfully written!
Article 14687 successfully written!
Article 14688 successfully written!
Article 14689 successfully written!
Article 14690 successfully written!
Article 14691 successfully w

Article 14888 successfully written!
Article 14889 successfully written!
Article 14890 successfully written!
Article 14891 successfully written!
Article 14892 successfully written!
Article 14893 successfully written!
Article 14894 successfully written!
Article 14895 successfully written!
Article 14896 successfully written!
Article 14897 successfully written!
Article 14898 successfully written!
Article 14899 successfully written!
Directory 'page_298.html' created
Article 14900 successfully written!
Article 14901 successfully written!
Article 14902 successfully written!
Article 14903 successfully written!
Article 14904 successfully written!
Article 14905 successfully written!
Article 14906 successfully written!
Article 14907 successfully written!
Article 14908 successfully written!
Article 14909 successfully written!
Article 14910 successfully written!
Article 14911 successfully written!
Article 14912 successfully written!
Article 14913 successfully written!
Article 14914 successfully wri

Article 15112 successfully written!
Article 15113 successfully written!
Article 15114 successfully written!
Article 15115 successfully written!
Article 15116 successfully written!
Article 15117 successfully written!
Article 15118 successfully written!
Article 15119 successfully written!
Article 15120 successfully written!
Article 15121 successfully written!
Article 15122 successfully written!
Article 15123 successfully written!
Article 15124 successfully written!
Article 15125 successfully written!
Article 15126 successfully written!
Article 15127 successfully written!
Article 15128 successfully written!
Article 15129 successfully written!
Article 15130 successfully written!
Article 15131 successfully written!
Article 15132 successfully written!
Article 15133 successfully written!
Article 15134 successfully written!
Article 15135 successfully written!
Article 15136 successfully written!
Article 15137 successfully written!
Article 15138 successfully written!
Article 15139 successfully w

Article 15336 successfully written!
Article 15337 successfully written!
Article 15338 successfully written!
Article 15339 successfully written!
Article 15340 successfully written!
Article 15341 successfully written!
Article 15342 successfully written!
Article 15343 successfully written!
Article 15344 successfully written!
Article 15345 successfully written!
Article 15346 successfully written!
Article 15347 successfully written!
Article 15348 successfully written!
Article 15349 successfully written!
Directory 'page_307.html' created
Article 15350 successfully written!
Article 15351 successfully written!
Article 15352 successfully written!
Article 15353 successfully written!
Article 15354 successfully written!
Article 15355 successfully written!
Article 15356 successfully written!
Article 15357 successfully written!
Article 15358 successfully written!
Article 15359 successfully written!
Article 15360 successfully written!
Article 15361 successfully written!
Article 15362 successfully wri

Article 15560 successfully written!
Article 15561 successfully written!
Article 15562 successfully written!
Article 15563 successfully written!
Article 15564 successfully written!
Article 15565 successfully written!
Article 15566 successfully written!
Article 15567 successfully written!
Article 15568 successfully written!
Article 15569 successfully written!
Article 15570 successfully written!
Article 15571 successfully written!
Article 15572 successfully written!
Article 15573 successfully written!
Article 15574 successfully written!
Article 15575 successfully written!
Article 15576 successfully written!
Article 15577 successfully written!
Article 15578 successfully written!
Article 15579 successfully written!
Article 15580 successfully written!
Article 15581 successfully written!
Article 15582 successfully written!
Article 15583 successfully written!
Article 15584 successfully written!
Article 15585 successfully written!
Article 15586 successfully written!
Article 15587 successfully w

Article 15784 successfully written!
Article 15785 successfully written!
Article 15786 successfully written!
Article 15787 successfully written!
Article 15788 successfully written!
Article 15789 successfully written!
Article 15790 successfully written!
Article 15791 successfully written!
Article 15792 successfully written!
Article 15793 successfully written!
Article 15794 successfully written!
Article 15795 successfully written!
Article 15796 successfully written!
Article 15797 successfully written!
Article 15798 successfully written!
Article 15799 successfully written!
Directory 'page_316.html' created
Article 15800 successfully written!
Article 15801 successfully written!
Article 15802 successfully written!
Article 15803 successfully written!
Article 15804 successfully written!
Article 15805 successfully written!
Article 15806 successfully written!
Article 15807 successfully written!
Article 15808 successfully written!
Article 15809 successfully written!
Article 15810 successfully wri

Article 16007 successfully written!
Article 16008 successfully written!
Article 16009 successfully written!
Article 16010 successfully written!
Article 16011 successfully written!
Article 16012 successfully written!
Article 16013 successfully written!
Article 16014 successfully written!
Article 16015 successfully written!
Article 16016 successfully written!
Article 16017 successfully written!
Article 16018 successfully written!
Article 16019 successfully written!
Article 16020 successfully written!
Article 16021 successfully written!
Article 16022 successfully written!
Article 16023 successfully written!
Article 16024 successfully written!
Article 16025 successfully written!
Article 16026 successfully written!
Article 16027 successfully written!
Article 16028 successfully written!
Article 16029 successfully written!
Article 16030 successfully written!
Article 16031 successfully written!
Article 16032 successfully written!
Article 16033 successfully written!
Article 16034 successfully w

Article 16232 successfully written!
Article 16233 successfully written!
Article 16234 successfully written!
Article 16235 successfully written!
Article 16236 successfully written!
Article 16237 successfully written!
Article 16238 successfully written!
Article 16239 successfully written!
Article 16240 successfully written!
Article 16241 successfully written!
Article 16242 successfully written!
Article 16243 successfully written!
Article 16244 successfully written!
Article 16245 successfully written!
Article 16246 successfully written!
Article 16247 successfully written!
Article 16248 successfully written!
Article 16249 successfully written!
Directory 'page_325.html' created
Article 16250 successfully written!
Article 16251 successfully written!
Article 16252 successfully written!
Article 16253 successfully written!
Article 16254 successfully written!
Article 16255 successfully written!
Article 16256 successfully written!
Article 16257 successfully written!
Article 16258 successfully wri

Article 16456 successfully written!
Article 16457 successfully written!
Article 16458 successfully written!
Article 16459 successfully written!
Article 16460 successfully written!
Article 16461 successfully written!
Article 16462 successfully written!
Article 16463 successfully written!
Article 16464 successfully written!
Article 16465 successfully written!
Article 16466 successfully written!
Article 16467 successfully written!
Article 16468 successfully written!
Article 16469 successfully written!
Article 16470 successfully written!
Article 16471 successfully written!
Article 16472 successfully written!
Article 16473 successfully written!
Article 16474 successfully written!
Article 16475 successfully written!
Article 16476 successfully written!
Article 16477 successfully written!
Article 16478 successfully written!
Article 16479 successfully written!
Article 16480 successfully written!
Article 16481 successfully written!
Article 16482 successfully written!
Article 16483 successfully w

Article 16680 successfully written!
Article 16681 successfully written!
Article 16682 successfully written!
Article 16683 successfully written!
Article 16684 successfully written!
Article 16685 successfully written!
Article 16686 successfully written!
Article 16687 successfully written!
Article 16688 successfully written!
Article 16689 successfully written!
Article 16690 successfully written!
Article 16691 successfully written!
Article 16692 successfully written!
Article 16693 successfully written!
Article 16694 successfully written!
Article 16695 successfully written!
Article 16696 successfully written!
Article 16697 successfully written!
Article 16698 successfully written!
Article 16699 successfully written!
Directory 'page_334.html' created
Article 16700 successfully written!
Article 16701 successfully written!
Article 16702 successfully written!
Article 16703 successfully written!
Article 16704 successfully written!
Article 16705 successfully written!
Article 16706 successfully wri

Article 16904 successfully written!
Article 16905 successfully written!
Article 16906 successfully written!
Article 16907 successfully written!
Article 16908 successfully written!
Article 16909 successfully written!
Article 16910 successfully written!
Article 16911 successfully written!
Article 16912 successfully written!
Article 16913 successfully written!
Article 16914 successfully written!
Article 16915 successfully written!
Article 16916 successfully written!
Article 16917 successfully written!
Article 16918 successfully written!
Article 16919 successfully written!
Article 16920 successfully written!
Article 16921 successfully written!
Article 16922 successfully written!
Article 16923 successfully written!
Article 16924 successfully written!
Article 16925 successfully written!
Article 16926 successfully written!
Article 16927 successfully written!
Article 16928 successfully written!
Article 16929 successfully written!
Article 16930 successfully written!
Article 16931 successfully w

Article 17128 successfully written!
Article 17129 successfully written!
Article 17130 successfully written!
Article 17131 successfully written!
Article 17132 successfully written!
Article 17133 successfully written!
Article 17134 successfully written!
Article 17135 successfully written!
Article 17136 successfully written!
Article 17137 successfully written!
Article 17138 successfully written!
Article 17139 successfully written!
Article 17140 successfully written!
Article 17141 successfully written!
Article 17142 successfully written!
Article 17143 successfully written!
Article 17144 successfully written!
Article 17145 successfully written!
Article 17146 successfully written!
Article 17147 successfully written!
Article 17148 successfully written!
Article 17149 successfully written!
Directory 'page_343.html' created
Article 17150 successfully written!
Article 17151 successfully written!
Article 17152 successfully written!
Article 17153 successfully written!
Article 17154 successfully wri

Article 17352 successfully written!
Article 17353 successfully written!
Article 17354 successfully written!
Article 17355 successfully written!
Article 17356 successfully written!
Article 17357 successfully written!
Article 17358 successfully written!
Article 17359 successfully written!
Article 17360 successfully written!
Article 17361 successfully written!
Article 17362 successfully written!
Article 17363 successfully written!
Article 17364 successfully written!
Article 17365 successfully written!
Article 17366 successfully written!
Article 17367 successfully written!
Article 17368 successfully written!
Article 17369 successfully written!
Article 17370 successfully written!
Article 17371 successfully written!
Article 17372 successfully written!
Article 17373 successfully written!
Article 17374 successfully written!
Article 17375 successfully written!
Article 17376 successfully written!
Article 17377 successfully written!
Article 17378 successfully written!
Article 17379 successfully w

Article 17577 successfully written!
Article 17578 successfully written!
Article 17579 successfully written!
Article 17580 successfully written!
Article 17581 successfully written!
Article 17582 successfully written!
Article 17583 successfully written!
Article 17584 successfully written!
Article 17585 successfully written!
Article 17586 successfully written!
Article 17587 successfully written!
Article 17588 successfully written!
Article 17589 successfully written!
Article 17590 successfully written!
Article 17591 successfully written!
Article 17592 successfully written!
Article 17593 successfully written!
Article 17594 successfully written!
Article 17595 successfully written!
Article 17596 successfully written!
Article 17597 successfully written!
Article 17598 successfully written!
Article 17599 successfully written!
Directory 'page_352.html' created
Article 17600 successfully written!
Article 17601 successfully written!
Article 17602 successfully written!
Article 17603 successfully wri

Article 17801 successfully written!
Article 17802 successfully written!
Article 17803 successfully written!
Article 17804 successfully written!
Article 17805 successfully written!
Article 17806 successfully written!
Article 17807 successfully written!
Article 17808 successfully written!
Article 17809 successfully written!
Article 17810 successfully written!
Article 17811 successfully written!
Article 17812 successfully written!
Article 17813 successfully written!
Article 17814 successfully written!
Article 17815 successfully written!
Article 17816 successfully written!
Article 17817 successfully written!
Article 17818 successfully written!
Article 17819 successfully written!
Article 17820 successfully written!
Article 17821 successfully written!
Article 17822 successfully written!
Article 17823 successfully written!
Article 17824 successfully written!
Article 17825 successfully written!
Article 17826 successfully written!
Article 17827 successfully written!
Article 17828 successfully w

Article 18026 successfully written!
Article 18027 successfully written!
Article 18028 successfully written!
Article 18029 successfully written!
Article 18030 successfully written!
Article 18031 successfully written!
Article 18032 successfully written!
Article 18033 successfully written!
Article 18034 successfully written!
Article 18035 successfully written!
Article 18036 successfully written!
Article 18037 successfully written!
Article 18038 successfully written!
Article 18039 successfully written!
Article 18040 successfully written!
Article 18041 successfully written!
Article 18042 successfully written!
Article 18043 successfully written!
Article 18044 successfully written!
Article 18045 successfully written!
Article 18046 successfully written!
Article 18047 successfully written!
Article 18048 successfully written!
Article 18049 successfully written!
Directory 'page_361.html' created
Article 18050 successfully written!
Article 18051 successfully written!
Article 18052 successfully wri

Article 18251 successfully written!
Article 18252 successfully written!
Article 18253 successfully written!
Article 18254 successfully written!
Article 18255 successfully written!
Article 18256 successfully written!
Article 18257 successfully written!
Article 18258 successfully written!
Article 18259 successfully written!
Article 18260 successfully written!
Article 18261 successfully written!
Article 18262 successfully written!
Article 18263 successfully written!
Article 18264 successfully written!
Article 18265 successfully written!
Article 18266 successfully written!
Article 18267 successfully written!
Article 18268 successfully written!
Article 18269 successfully written!
Article 18270 successfully written!
Article 18271 successfully written!
Article 18272 successfully written!
Article 18273 successfully written!
Article 18274 successfully written!
Article 18275 successfully written!
Article 18276 successfully written!
Article 18277 successfully written!
Article 18278 successfully w

Article 18475 successfully written!
Article 18476 successfully written!
Article 18477 successfully written!
Article 18478 successfully written!
Article 18479 successfully written!
Article 18480 successfully written!
Article 18481 successfully written!
Article 18482 successfully written!
Article 18483 successfully written!
Article 18484 successfully written!
Article 18485 successfully written!
Article 18486 successfully written!
Article 18487 successfully written!
Article 18488 successfully written!
Article 18489 successfully written!
Article 18490 successfully written!
Article 18491 successfully written!
Article 18492 successfully written!
Article 18493 successfully written!
Article 18494 successfully written!
Article 18495 successfully written!
Article 18496 successfully written!
Article 18497 successfully written!
Article 18498 successfully written!
Article 18499 successfully written!
Directory 'page_370.html' created
Article 18500 successfully written!
Article 18501 successfully wri

Article 18700 successfully written!
Article 18701 successfully written!
Article 18702 successfully written!
Article 18703 successfully written!
Article 18704 successfully written!
Article 18705 successfully written!
Article 18706 successfully written!
Article 18707 successfully written!
Article 18708 successfully written!
Article 18709 successfully written!
Article 18710 successfully written!
Article 18711 successfully written!
Article 18712 successfully written!
Article 18713 successfully written!
Article 18714 successfully written!
Article 18715 successfully written!
Article 18716 successfully written!
Article 18717 successfully written!
Article 18718 successfully written!
Article 18719 successfully written!
Article 18720 successfully written!
Article 18721 successfully written!
Article 18722 successfully written!
Article 18723 successfully written!
Article 18724 successfully written!
Article 18725 successfully written!
Article 18726 successfully written!
Article 18727 successfully w

Article 18924 successfully written!
Article 18925 successfully written!
Article 18926 successfully written!
Article 18927 successfully written!
Article 18928 successfully written!
Article 18929 successfully written!
Article 18930 successfully written!
Article 18931 successfully written!
Article 18932 successfully written!
Article 18933 successfully written!
Article 18934 successfully written!
Article 18935 successfully written!
Article 18936 successfully written!
Article 18937 successfully written!
Article 18938 successfully written!
Article 18939 successfully written!
Article 18940 successfully written!
Article 18941 successfully written!
Article 18942 successfully written!
Article 18943 successfully written!
Article 18944 successfully written!
Article 18945 successfully written!
Article 18946 successfully written!
Article 18947 successfully written!
Article 18948 successfully written!
Article 18949 successfully written!
Directory 'page_379.html' created
Article 18950 successfully wri